core/benches/receive_and_buffer.rs

```diff
@@ -0,0 +1,278 @@
+    for keypair in account_keypairs.iter() {
+        genesis_config.add_account(keypair.pubkey(), AccountSharedData::new(10000, 0, &owner));
+    }
+    account_keypairs
+}
+
+/// Structure that returns correct provided blockhash or some incorrect hash
+/// with given probability.
+pub struct FaultyBlockhash {
+    blockhash: Hash,
+    probability_invalid_blockhash: f64,
+}
+
+impl FaultyBlockhash {
+    /// Create a new faulty hash generator
+    pub fn new(blockhash: Hash, probability_invalid_blockhash: f64) -> Self {
+        Self {
+            blockhash,
+            probability_invalid_blockhash,
+        }
+    }
+
+    pub fn get<R: Rng>(&self, rng: &mut R) -> Hash {
+        if rng.gen::<f64>() < self.probability_invalid_blockhash {
+            Hash::default()
+        } else {
+            self.blockhash
+        }
+    }
+}
+
+fn generate_transactions(
+    num_txs: usize,
+    bank: Arc<Bank>,
+    fee_payers: &[Keypair],
+    num_instructions_per_tx: usize,
+    probability_invalid_blockhash: f64,
+) -> BankingPacketBatch {
+    assert!(num_instructions_per_tx <= MAX_INSTRUCTIONS_PER_TRANSACTION);
+    let blockhash = FaultyBlockhash::new(bank.last_blockhash(), probability_invalid_blockhash);
+
+    let mut rng = rand::thread_rng();
+
+    let mut fee_payers = fee_payers.iter().cycle();
+
+    let txs: Vec<VersionedTransaction> = (0..num_txs)
+        .map(|_| {
+            let fee_payer = fee_payers.next().unwrap();
+            let program_id = Pubkey::new_unique();
+
+            let mut instructions = Vec::with_capacity(num_instructions_per_tx);
+            // Experiments with different distributions didn't show much of the effect on the performance.
+            let compute_unit_price = rng.gen_range(0..1000);
+            instructions.push(ComputeBudgetInstruction::set_compute_unit_price(
+                compute_unit_price,
+            ));
+            for _ in 0..num_instructions_per_tx - 1 {
+                instructions.push(Instruction::new_with_bytes(
+                    program_id,
+                    &[0],
+                    vec![AccountMeta {
+                        pubkey: fee_payer.pubkey(),
+                        is_signer: true,
+                        is_writable: true,
+                    }],
+                ));
+            }
+            VersionedTransaction::try_new(
+                VersionedMessage::Legacy(Message::new_with_blockhash(
+                    &instructions,
+                    Some(&fee_payer.pubkey()),
+                    &blockhash.get(&mut rng),
+                )),
+                &[&fee_payer],
+            )
+            .unwrap()
+        })
+        .collect();
+
+    let packets_batches = BankingPacketBatch::new(to_packet_batches(&txs, NUM_PACKETS));
+    return packets_batches;
+}
+
+// TODO(klykov): maybe not needed anylonger if add new to the TransactionViewReceiveAndBuffer?
```

Comment:
this is done, correct?

Reply:
I have doubts there: instead of having this code I can just add `TransactionViewReceiveAndBuffer::new` and this will simplify benchmark for the price of adding `new` which is sort of not really of a big value but also doesn't hurt. Maybe adding `new` is ok?